### ISSUE TITLE
Delay when saving files in projects with tasks.json

### ISSUE THREAD

**Issue by gshaw** [Microsoft/vscode#19945]
- VSCode Version: 1.9.0
- OS Version: macOS Sierra 10.12.3

Steps to Reproduce:

1. Create new directory for a minimal test project.
```sh
mkdir test
cd test
mkdir .vscode
echo // Empty File > .vscode/tasks.json
echo Test > test.txt
```

2. Open minimal test project
```
code --disable-extensions . 
```

3. Edit `test.txt` file (e.g., add a blank line) and save changes (File > Save)

Expected: File to immediately save as it did with 1.8.1 (and verified after downgrading to 1.8.1)

Actual: File saves but with approximate 1 second delay.

Workarounds:
* Remove the `tasks.json`
* Revert to [1.8.1](https://vscode-update.azurewebsites.net/1.8.1/darwin/stable)

**Comment by bpasero** [comment]
I wonder if this issue is being caused by an installed extension. Can you try to run VS Code without extensions? From the command line (NOT the integrated terminal in Code), execute: `code --disable-extensions` and try your steps again to see if it reproduces. If you see it is an issue with the extension, please file it against the extension repository itself.

**Comment by gshaw** [comment]
That is a great idea.  I tried it and it still had the slowdown but it got me thinking that it might have been something in my project so I managed to narrow down the test case to make it reproducible.

**Comment by gshaw** [comment]
I've updated the issue title and description with the steps to reproduce.  It seems to be triggered by the presence of a `.vscode/tasks.json` file.

**Comment by bpasero** [comment]
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by gshaw** [comment]
Attached is a Gist with the Terminal console log and the web console log.  I ran it with extensions disabled to ensure I wasn't logging any extra junk from my extensions.

https://gist.github.com/gshaw/0d76fae2a6fa147501d8a740982390a8

Unfortunately I'm not seeing anything of much help to move this along.  The only thing I did notice was the log message for `ipc.cp.ts:152 [IPC Library: Watcher] [CHANGED] /Users/gerry/test/test.txt` didn't appear in the log until after the delay between triggering a save and the save actually happening.

Is anybody else able to duplicate the problem? I'll try on Monday (Pacific Time) when I have access to other Macs and comment back here.

**Comment by bpasero** [comment]
@gshaw this seems to be https://github.com/Microsoft/vscode/issues/19773 and I kindly ask you to maybe try what I suggested (set "break on error" and see if you are able to hit the breakpoint where the error happens).

**Comment by gshaw** [comment]
I didn't hit any breakpoints during save but looking at #19733 this issue seems like it's resolved.  If you need any other info let me know.  The fast response and great communication is much appreciated.

**Comment by bpasero** [comment]
Yeah we fixed it and will backport it to a 1.9.1 release.
